app.rb

```diff
@@ -0,0 +1,92 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  def initialize
+    puts 'Welcome to School Library App!'
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each do |object|
+      puts "Title: '#{object.title}', Author: #{object.author}"
+    end
+  end
+
+  def list_people
+    @people.each do |object|
+      puts "[#{object.class}] Name: #{object.name}, ID: #{object.id}, Age: #{object.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    option_person = gets.chomp.to_i
+    case option_person
+    when 1
+      print 'Age: '
+      person_age = gets.chomp.to_i
+      print 'Name: '
+      person_name = gets.chomp.to_s
+      print 'Has parent permission? [Y/N]: '
+      person_permission = gets.chomp.downcase
+      if person_permission.capitalize == 'N'
+        @people.push(Student.new(age: person_age, name: person_name, parent_permission: false))
+      else
+        @people.push(Student.new(age: person_age, name: person_name))
+      end
+      puts 'Student created successfully'
+    when 2
+      print 'Age: '
+      person_age = gets.chomp.to_i
+      print 'Name: '
+      person_name = gets.chomp.to_s
+      print 'Specialization: '
+      person_specialization = gets.chomp.to_s
+      @people.push(Teacher.new(person_specialization, person_name, person_age))
+      puts 'Teacher created successfully'
+    else
+      puts 'Error: Invalid number, try again'
+      create_person
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    book_title = gets.chomp.to_s
+    print 'Author: '
+    book_author = gets.chomp.to_s
+    @books.push(Book.new(book_title, book_author))
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index do |object, index|
+      puts "#{index}) Title: '#{object.title}', Author: #{object.author}"
+    end
+    book_number = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |object, index|
+      puts "#{index}) [#{object.class}] Name: #{object.name}, ID: #{object.id}, Age: #{object.age}"
+    end
+    person_number = gets.chomp.to_i
+    print 'Date: '
+    date = gets.chomp.to_s
+    @rentals.push(Rental.new(@books[book_number], @people[person_number], date))
+    puts 'Rental created successfully'
```

Comment:
- [ ] There's stilll an error while creating your rentals. Firstly, the prompt does not provide "List of numbers" as you mentioned, an secondly, kindly review undefined `rentals` method called in your rentals file which raised the error below :

### Error:
![image](https://github.com/enmanuelbayen/OOP-school-library/assets/50990925/839e19b1-3d07-439e-9131-2b52e1e03c11)

Reply:
Hi! @Shedrack-Sunday I can see that you try to create the rental without creating the person and book before. that is the reason for the error.  I can try to create a conditional to prevent this but is not in the req just to clarify.  
I will also put an example here that the code works with the person and book create:

### create teacher and student
![Screenshot 2023-07-28 162703](https://github.com/enmanuelbayen/OOP-school-library/assets/121642946/72dc8969-95a3-4c28-b6ac-59d007681b45)
### list of person and create book and show the list
![Screenshot 2023-07-28 162719](https://github.com/enmanuelbayen/OOP-school-library/assets/121642946/21196110-e385-477a-bc0f-2c5e9e0f0944)
### Create rental and show list of rental
![Screenshot 2023-07-28 162732](https://github.com/enmanuelbayen/OOP-school-library/assets/121642946/889831d4-dfb9-49e3-84ff-95a7f2c0d655)

Reply:
Hello @enmanuelbayen , there was an existing person with an `id` of `630`, and a `no.` 0. That was ideally meant to work. Though  i noticed you've been approved 👍